racket/collects/racket/private/hash.rkt

```diff
@@ -93,6 +93,22 @@
        [(hash-eqv? table) (make-hasheqv)]
        [(hash-eq? table) (make-hasheq)])]))
 
+  (define (hash-map/copy table f)
+    (unless (hash? table)
+      (raise-argument-error 'hash-map/copy "hash?" table))
+    (cond
+     [(immutable? table)
+      (for/fold ([acc (hash-copy-clear table)])
+                ([(k1 v1) (in-hash table)])
```

Comment:
Not sure if it matters, but there's a family of `in-XXX-hash`, like `in-immutable-hash`, that could make iteration faster.

Reply:
I can use `in-immutable-hash`, but I don't want to use the others because it's simpler to have the same code path handle mutable, weak, and ephemeron hashes.